### PR TITLE
fix(compiler): panic when trying to reference an unknown field

### LINKED ISSUE #2602: Compiler panics on unknown field
### I tried this:

```ts (wing)
bring cloud;

let api = new cloud.Api();

api.get("/fib/{n}", inflight (req: cloud.ApiRequest): cloud.ApiResponse => {
  let n = num.pa.vars.get("n"); // this lines causes panic
  return cloud.ApiResponse { status: 200, body: "hello" };
});
```

### This happened:

compiler paniced

### I expected this:

not to panic

### Is there a workaround?

_No response_

### Component

Compiler

### Wing Version

_No response_

### Wing Console Version

_No response_

### Node.js Version

_No response_

### Platform(s)

_No response_

### Anything else?

_No response_

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

### LINKED ISSUE #2877: Calling a static method through non-existing submodules is possible
### I tried this:

```js
std.xoo.hoo.zoo.boo.String.fromJson("hello");
```

### This happened:

It works!

### I expected this:

An error saying that this type cannot be found.

### Is there a workaround?

_No response_

### Component

Compiler

### Wing Version

_No response_

### Wing Console Version

_No response_

### Node.js Version

_No response_

### Platform(s)

_No response_

### Anything else?

_No response_

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

### PR DESCRIPTION
**Pull request by eladb**
Fixes #2602 by avoiding the creation of namespaces blindly, even if a type doesn't exist in that namespace.

Additionally fixes #2877 which was caused by mistakenly associating its symbol environment with the parent submodule, but that's not actually what we want. If I couldn't find `String` in `std.foo`, I don't want to look up string in `std`. The diagnostics emitted in this case is still weird, so I created #2876 to address this.

## Checklist

- [x] Title matches [Winglang's style guide](https://docs.winglang.io/contributors/pull_requests#how-are-pull-request-titles-formatted)
- [x] Description explains motivation and solution
- [x] Tests added (always)
- [x] Docs updated (only required for features)
- [x] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

**Comment by eladb** [comment]
@Chriscbr can you take a look?

**Review by Chriscbr — APPROVED** [review]

File: examples/tests/invalid/unknown_field.w [hunk 1]
```diff
@@ -0,0 +1,2 @@
+std.String.a.b.c.fromJson();
+//         ^ No member "a" in class "String"
```
[the remark this answers is not in the document]
Comment (Chriscbr) [reply examples/tests/invalid/unknown_field.w:2]: also does this only affect `std` or does `cloud.Bucket.a.b.c.addFile` will produce a similar error?

[the remark this answers is not in the document]
Comment (eladb) [reply examples/tests/invalid/unknown_field.w:2]: Mmmm. not sure I agree about not having access to `std` (let's discuss of course). So in the meantime, I'll leave it like this. But it is easy to use `cloud` as an alternative of course.

[the remark this answers is not in the document]
Comment (eladb) [reply examples/tests/invalid/unknown_field.w:2]: Same error for any namespace

**Pull request merged**